### DISCUSSION TITLE
What the difference

### DISCUSSION THREAD

**Discussion by Prathapkarumanchi** [konstaui/konsta#107] [General]
Hi,

What are the differences between konstaui vs framework7. 
Tailwincss is the only difference?

**Comment by aabluedragon** [comment]
I searched everywhere and found no answer, I think most users who stumble upon both libraries ask the same question, so far the only difference I found was that KonstaUI makes use of tailwind, however it stil begs the question: which library to use, and when, f7 or konsta?
I've only been using f7, with svelte, but usage looks pretty much the same from the docs, except the tailwind config.
I think it would serve everyone better if we had a clear answer.

**Comment by hakimzulkufli** [comment]
Konsta UI is a UI library that provides pre-styled UI components designed specifically for mobile applications. Because it only provides UI components, you can use it wherever and however you want.

Framework7 is more than that. It provides (almost) full functionalities for you to build a mobile application. It is powered by the supported UI framework, such as Svelte and React. Think of Framework7 like React Native or Expo or Ionic.

**Comment by jkhaui** [comment]
IMO, perception is key here. I would describe Konsta's first-class tailwind integration as its major selling point, as opposed to being the "only" difference with framework7.

From what I understand, framework7 is nearly a decade old. I'm sure it's great at what it does (I've never used it myself) but as with all things in tech, sometimes projects must re-invent themselves as developer norms, tools, platform standards, etc. evolve. To this end, it's inarguable that tailwind has emerged as the dominant CSS framework today (with over 6.5 million weekly installs). Its simplicity has resulted in rapid adoption by frontend devs and Konsta does an excellent job in leveraging this fact to spin off a "lite" adaptive UI library which is much easier to consume than heavier frameworks like framework or Ionic.

Another take is that a pure UI lib like Konsta is much more portable and thus has more use-cases than a "parent" framework which also includes an opinionated router, renderer, etc. Currently I'm working on a bespoke framework with its own custom routing system which makes integrating framework7 impossible... However, in this instance, Konsta is the perfect fit and it's been an almost seamless integration so far.

So if my rationale is correct then Konsta was an excellent strategic move by the creator (who's also been a prolific contributor to "web native" OS repos). I hope the community helps to grow it 😄

**Reply by aabluedragon** [reply]
Well said.
I found myself porting my Svelte+Framework7 app to SvelteKit+Konsta due to all sorts of routing issues and limitations on framework7, and other missing features that SvelteKit has.
There were some drawbacks during this migration, e.g. for managing programmatic dialogs (avoiding one template per popup), I had to use custom logic, and in the case of SvelteKit, I used this wrapper to define my dialogs:
https://github.com/vnphanquang/svelte-put/tree/main/packages/misc/modal

**Comment by Derewith** [comment]
Hey guys, was wondering if there are still big differences related in 2026
